mongodb/src/main/java/org/hibernate/ogm/datastore/mongodb/type/impl/MongoDBTypeContributor.java

```diff
@@ -20,5 +20,6 @@
 	@Override
 	public void contribute(TypeContributions typeContributions, ServiceRegistry serviceRegistry) {
 		typeContributions.contributeType( new StringAsObjectIdType() );
+		typeContributions.contributeType( new BytesAsBsonBinaryType() );
```

Comment:
@DavideD I rechecked.
______________________
java.lang.ClassCastException: org.bson.types.Binary cannot be cast to [B
_____________________________________